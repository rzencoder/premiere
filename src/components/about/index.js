import React from "react";
import { Container, Item, Divider, Text, Title, Icon } from "./styles/about";

export default function About({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

About.Item = function AboutItem({ children, ...restProps }) {
  return <Item {...restProps}>{children}</Item>;
};

About.Divider = function AboutDivider({ children, ...restProps }) {
  return <Divider {...restProps} />;
};

About.Title = function AboutTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

About.Text = function AboutText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

About.Icon = function AboutIcon({ src, ...restProps }) {
  return <Icon {...restProps} src={src} />;
};
