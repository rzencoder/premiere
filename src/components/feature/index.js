import React from "react";
import {
  Background,
  Container,
  Overlay,
  TextContainer,
  Title,
  Text,
  Button,
  Details,
  SubText,
  Rating,
  AgeRating,
  Year,
} from "./styles/feature";

export default function Feature({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Feature.Background = function FeatureBackground({ ...restProps }) {
  return <Background {...restProps} />;
};

Feature.Overlay = function FeatureOverlay({ ...restProps }) {
  return <Overlay {...restProps} />;
};

Feature.TextContainer = function FeatureTextContainer({
  children,
  ...restProps
}) {
  return <TextContainer {...restProps}>{children}</TextContainer>;
};

Feature.Title = function FeatureTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Feature.Text = function FeatureText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Feature.Button = function FeatureButton({ children, ...restProps }) {
  return <Button {...restProps}>{children}</Button>;
};

Feature.Details = function FeatureDetails({ children, ...restProps }) {
  return <Details {...restProps}>{children}</Details>;
};

Feature.SubText = function FeatureSubText({ children, ...restProps }) {
  return <SubText {...restProps}>{children}</SubText>;
};

Feature.AgeRating = function FeatureAgeRating({ ...restProps }) {
  return <AgeRating {...restProps} />;
};

Feature.Year = function FeatureYear({ children, ...restProps }) {
  return (
    <Year {...restProps}>
      <div>{children}</div>
    </Year>
  );
};

Feature.Rating = function FeatureRating({ children, ...restProps }) {
  return (
    <Rating {...restProps}>
      <div>{children}</div>
    </Rating>
  );
};
