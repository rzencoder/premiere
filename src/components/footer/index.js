import React from "react";
import {
  Container,
  Text,
  Subtitle,
  Section,
  Link,
  Divider,
  Column,
  Break,
} from "./styles/footer";

export default function Footer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Footer.Text = function FooterText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Footer.Subtitle = function FooterSubtitle({ children, ...restProps }) {
  return <Subtitle {...restProps}>{children}</Subtitle>;
};

Footer.Link = function FooterLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Footer.Divider = function FooterDivider({ ...restProps }) {
  return <Divider {...restProps} />;
};

Footer.Section = function FooterSection({ children, ...restProps }) {
  return <Section {...restProps}>{children}</Section>;
};

Footer.Column = function FooterColumn({ children, ...restProps }) {
  return <Column {...restProps}>{children}</Column>;
};

Footer.Break = function FooterBreak({ ...restProps }) {
  return <Break {...restProps} />;
};
