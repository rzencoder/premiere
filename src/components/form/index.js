import React from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Container,
  Title,
  Content,
  Label,
  Input,
  Submit,
  Frame,
  Text,
  Link,
  Error,
} from "./styles/form";

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Frame = function FormFrame({ children, ...restProps }) {
  return (
    <Frame {...restProps}>
      <div>{children}</div>
    </Frame>
  );
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Content = function FormContent({ children, ...restProps }) {
  return <Content {...restProps}>{children}</Content>;
};

Form.Label = function FormLabel({ children, ...restProps }) {
  return <Label {...restProps}>{children}</Label>;
};

Form.Input = function FormInput({ ...restProps }) {
  return <Input {...restProps} />;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.Link = function FormLink({ to, children, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Link {...restProps}>{children}</Link>
    </ReactRouterLink>
  );
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};
