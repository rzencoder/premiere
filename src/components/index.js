export { default as Navigation } from "./navigation";
export { default as Jumbotron } from "./jumbotron";
export { default as About } from "./about";
export { default as Footer } from "./footer";
export { default as Form } from "./form";
