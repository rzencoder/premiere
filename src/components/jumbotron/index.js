import React from "react";
import {
  Background,
  Container,
  Overlay,
  Group,
  Title,
  Text,
  CallToAction,
} from "./styles/jumbotron";
import { Link as ReactRouterLink } from "react-router-dom";

export default function Jumbotron({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Jumbotron.Overlay = function JumbotronOverlay({ ...restProps }) {
  return <Overlay {...restProps} />;
};

Jumbotron.Group = function JumbotronGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};
Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.Text = function JumbotronText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Jumbotron.CallToAction = function JumbotronCallToAction({
  to,
  children,
  ...restProps
}) {
  return (
    <ReactRouterLink to={to}>
      <CallToAction {...restProps}>{children}</CallToAction>
    </ReactRouterLink>
  );
};

Jumbotron.Background = function JumbotronBackground({ ...restProps }) {
  return <Background {...restProps} />;
};
