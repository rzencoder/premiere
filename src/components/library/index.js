import React from "react";
import {
  Container,
  Image,
  Button,
  Divider,
  CarouselContainer,
  CarouselGroup,
  CarouselButton,
  Selector,
  SelectorItem,
  CarouselOverlay,
  Panel,
} from "./styles/library";

export default function Library({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Library.Button = function LibraryButton({ children, ...restProps }) {
  return <Button {...restProps}>{children}</Button>;
};

Library.Divider = function LibraryDivider({ ...restProps }) {
  return <Divider {...restProps} />;
};

Library.Image = function LibraryImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Library.CarouselContainer = function LibraryCarouselContainer({
  children,
  ...restProps
}) {
  return <CarouselContainer {...restProps}>{children}</CarouselContainer>;
};

Library.CarouselGroup = function LibraryCarouselGroup({
  children,
  ...restProps
}) {
  return <CarouselGroup {...restProps}>{children}</CarouselGroup>;
};

Library.CarouselButton = function LibraryCarouselButton({
  children,
  ...restProps
}) {
  return <CarouselButton {...restProps}>{children}</CarouselButton>;
};

Library.Selector = function LibrarySelector({ children, ...restProps }) {
  return (
    <Selector {...restProps}>
      <div>{children}</div>
    </Selector>
  );
};

Library.SelectorItem = function LibrarySelectorItem({
  children,
  ...restProps
}) {
  return <SelectorItem {...restProps}>{children}</SelectorItem>;
};

Library.CarouselOverlay = function LibraryCarouselOverlay({ ...restProps }) {
  return <CarouselOverlay {...restProps} />;
};

Library.Panel = function LibraryPanel({ children, ...restProps }) {
  return (
    <Panel {...restProps}>
      <div>{children}</div>
    </Panel>
  );
};
