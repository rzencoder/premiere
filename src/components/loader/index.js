import React from "react";
import { Container, Spinner } from "./styles/loader";

export default function Loader({ ...restProps }) {
  return (
    <Container data-testid="loader">
      <Spinner {...restProps} />;
    </Container>
  );
}
