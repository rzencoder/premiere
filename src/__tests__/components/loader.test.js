import React from "react";
import { render } from "@testing-library/react";
import { Loader } from "../../components";

describe("<Loader/>", () => {
  it("will render <Loader/>", () => {
    const { container, getByTestId } = render(<Loader />);

    expect(getByTestId("loader")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
