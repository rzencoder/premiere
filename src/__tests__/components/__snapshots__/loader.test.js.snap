// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader/> will render <Loader/> 1`] = `
<div
  class="loader__Container-sc-1l4u7o8-0 fKQJKh"
  data-testid="loader"
>
  <div
    class="loader__Spinner-sc-1l4u7o8-1 bDmmAB"
  />
  ;
</div>
`;
